packages/google_adsense/lib/src/adsense_stub.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'ad_unit_widget.dart';
+
+/// Main class to work with the library
+class Adsense {
+  /// Returns a singleton instance of Adsense library public interface
+  factory Adsense() => _instance ?? Adsense._internal();
+
+  Adsense._internal() {
+    _instance = this;
+  }
+
+  static Adsense? _instance = Adsense._internal();
+
+  /// Initialization API. Should be called ASAP, ideally in the main method of your app.
+  void initialize(String adClient) {
+    throw UnsupportedError('Only supported on web');
```

Comment:
Since we always throw when calling `initialize` outside of the web, I'm not sure it makes sense to have this split API. I think this should be a pure web package, and users would need to use it by conditionally importing it wherever they need it, for example having a:

```
Widget getAds(config);
```

That has separate implementations, and it calls this package when `kIsWeb`.

Removing the stubs from the conditional exports of this package would simplify it a lot!

(See the sign-in button example [here](https://github.com/flutter/packages/blob/main/packages/google_sign_in/google_sign_in/example/lib/src/sign_in_button.dart))

Reply:
@ditman thanks! In the case when the consuming app is cross-platform and ads are only implemented on Web, is there a way to import only web package? 

I couldn't find a way to do it, it seems like developers need to import "something" on other platforms, even if they only want web package. Happy to sync on that too